### PR TITLE
feat: adding GET /address-info

### PR DESCRIPTION
**Pull request by thr0wn**

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -747,6 +747,35 @@ walletRouter.post(
...
+/**
+ * GET request to obtain adress information
+ * For the docs, see api-docs.js
+ */
+ walletRouter.get(
+  '/address-info',
+  query('address').isString(),
+  query('token').isString().optional(),
+  async (req, res) => {
+    try {
```
Comment (msbrogli) [inline src/index.js:759]: Why do we have this try? Which exception are we expecting to handle?

File: src/index.js [hunk 2]
```diff
@@ -747,6 +747,35 @@ walletRouter.post(
   }
 );
 
+/**
+ * GET request to obtain adress information
+ * For the docs, see api-docs.js
```
Comment (msbrogli) [inline src/index.js:752]: Missing update in the api-docs.js.

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 2 above)
Comment (thr0wn) [reply src/index.js:752]: api-docs.js updated, thanks!

File: src/index.js (hunk 1 above)
Comment (thr0wn) [reply src/index.js:759]: If the specified address does not belong to the wallet an error will be fired. To correctly respond to the api caller, this try catch statement catches the error and format the return.

Comment (pedroferreira1) [reply src/index.js:759]: Maybe we should capture the specific error

**Review by pedroferreira1 — APPROVED** [review]

File: src/index.js (hunk 1 above)
Comment (msbrogli) [reply src/index.js:759]: Exactly. We must avoid capturing generic exceptions at all costs.

Comment (thr0wn) [reply src/index.js:759]: Done, thanks!

File: src/index.js [hunk 3]
```diff
@@ -747,6 +747,39 @@ walletRouter.post(
...
+    try {
+      // Query parameters validation
+      const validationResult = parametersValidation(req);
+      if (!validationResult.success) {
+        return res.status(400).json(validationResult);
+      }
+
+      const wallet = req.wallet;
+      const { address, token } = matchedData(req, { locations: ['query'] });
+      const result = wallet.getAddressInfo(address, { token });
```
Comment (msbrogli) [inline src/index.js:768]: I feel this is the place where the exception might happen. So, we should keep the try-catch block just around it.

File: src/index.js [hunk 4]
```diff
@@ -747,6 +747,39 @@ walletRouter.post(
...
+      res.send({
+        success: true,
+        ...result
+      });
+    } catch(error) {
+      if (error instanceof errors.AddressError) {
+        res.send({ success: false, error: error.message });
+      }
+
+      throw error;
```
Comment (msbrogli) [inline src/index.js:778]: Shouldn't it be in the `else` block? I mean, we shouldn't re-raise the exception if it was handled returning the error to the user, right?

**Review by msbrogli — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 3 above)
Comment (thr0wn) [reply src/index.js:768]: Done, thanks.

File: src/index.js (hunk 4 above)
Comment (thr0wn) [reply src/index.js]: Absolutely right, thanks.

**Review by msbrogli — APPROVED** [review]

**Review by msbrogli — APPROVED** [review]

**Pull request merged**